### PR TITLE
feat: phone verification

### PR DESCRIPTION
**Pull request by dgacevic**
https://bayes.atlassian.net/browse/RWPD-1388

### CODE REVIEW

File: brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift [hunk 1]
```diff
@@ -55,6 +55,11 @@ struct TextFieldModel: ViewModel {
     var showPasswordToggle: Bool = false
 }
 
+struct TextFieldReplace: ViewModel {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Views/Core/FETextField.swift:58]: can be removed

File: brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationViewController.swift [hunk 2]
```diff
@@ -89,12 +89,27 @@ class RegistrationConfirmationViewController: BaseTableViewController<AccountCoo
...
+        guard let section = sections.firstIndex(where: { $0.hashValue == Models.Section.help.hashValue }),
+              let cell = tableView.cellForRow(at: IndexPath(row: 0, section: section)) as? 
+                WrapperTableViewCell<MultipleButtonsView> else {
+            return
+        }
+        
+        cell.wrappedView.setEnabled(enabled: false, index: 0)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 10) {
+            cell.wrappedView.setEnabled(enabled: true, index: 0)
+        }
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationViewController.swift:102]: this logic can be inside FEButton?

File: brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationInteractor.swift [hunk 3]
```diff
@@ -334,6 +327,54 @@ class RegistrationConfirmationInteractor: NSObject, Interactor, RegistrationConf
         }
     }
     
+    private func executePhoneCodeResend() {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/RegistrationConfirmation/RegistrationConfirmationInteractor.swift:330]: please put a comment like` /// Phone code` above this line

File: brd-ios/breadwallet/src/Redesign/Scenes/PhoneNumber/PhoneNumberInteractor.swift [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    
+    func confirm(viewAction: PhoneNumberModels.Confirm.ViewAction) {
+        let phoneNumber = "+" + (dataStore?.country?.areaCode ?? "") + (dataStore?.phoneNumber ?? "")
+        
+        let requestData = PhoneNumberVerificationRequestData(phoneNumber: phoneNumber)
+        PhoneNumberVerificationWorker().execute(requestData: requestData) { [weak self] result in
+            switch result {
+            case .success(let verification):
+                if let errors = verification?.validationErrors, !errors.isEmpty {
```
Comment (kenanmamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/PhoneNumber/PhoneNumberInteractor.swift:52]: what a parkour with the errors disguised as response 200

Comment (dgacevic) [reply brd-ios/breadwallet/src/Redesign/Scenes/PhoneNumber/PhoneNumberInteractor.swift:52]: yup

**Review by kenanmamedoff — DISMISSED** [review]

**Review by kenanmamedoff — APPROVED** [review]

**Pull request merged**